app-modules/notification/src/Notifications/Channels/MailChannel.php

```diff
@@ -267,4 +283,20 @@ protected function getRecipients($notifiable, $notification, $message)
 
         return parent::getRecipients($notifiable, $notification, $message);
     }
+
+    protected function isAddressBounced(object $notifiable, string $recipientAddress): bool
```

Comment:
Does this need to check anything about the `notifiable`? Could it just be easier to check the bounced table directly to see if there are any emails listed in there that match?

Reply:
I think this is to allow password resets and other system emails sent to users to still go through, and only guard emails sent to prospects and students. It is the same for SMS. What do you think?